src/js/angular/explore/controllers.js

```diff
@@ -22,280 +27,277 @@ angular
+        ExploreRestService.getGraph($scope.resourceInfo, fileType.type)
+            .then((data) => {
+                if (fileType.type.indexOf('json') > -1) {
+                    data = JSON.stringify(data);
+                }
+                // TODO: Use bowser library to get the browser type
+                const ua = navigator.userAgent.toLowerCase();
+                if (ua.indexOf('safari') !== -1 && ua.indexOf('chrome') === -1) {
+                    window.open('data:attachment/csv;filename="statements.' + fileType.extension + '",' + encodeURIComponent(data), 'statements.' + fileType.extension);
+                } else {
+                    const file = new Blob([data], {type: fileType.type});
+                    saveAs(file, 'statements' + fileType.extension);
+                }
+            })
+            .catch((error) => {
+                toastr.error(getError(error), $translate.instant('common.error'));
+            })
+            .finally(() => {
+                toggleOntoLoader(false);
+            });
     };
 
-    $scope.changeRole = function (roleVar) {
-        $scope.role = roleVar;
-        if ($scope.role === 'context') {
-            $scope.inference = 'explicit';
+    $scope.changeRole = (role) => {
+        $scope.resourceInfo.role = role;
+        if ($scope.resourceInfo.role === RoleType.CONTEXT) {
+            $scope.resourceInfo.contextType = ContextTypes.EXPLICIT;
         }
         $scope.exploreResource();
     };
 
-    $scope.changeInference = function (inference) {
-        $scope.inference = inference;
+    $scope.changeInference = (contextTypeId) => {
+        $scope.resourceInfo.contextType = ContextType.getContextType(contextTypeId);
         $scope.exploreResource();
     };
 
-    $scope.copyToClipboardResult = function (uri) {
+    $scope.copyToClipboardResult = (uri) => {
         ModalService.openCopyToClipboardModal(uri);
     };
 
-    $scope.$on('language-changed', function (event, args) {
-        if (yasr && yasr.options) {
-            yasr.options.locale = args.locale;
-            yasr.changeLanguage(args.locale);
+    // =========================
+    // Private functions
+    // =========================
+    const initComponent = () => {
+        Promise.all([$jwtAuth.getPrincipal(), $repositories.getPrefixes($repositories.getActiveRepository())])
+            .then(([principal, usedPrefixesResponse]) => {
+                initResourceInfo();
+                setInferAndSameAs(principal);
+                usedPrefixes = usedPrefixesResponse;
+                $scope.loadResource();
+            })
+            .catch((error) => {
+                toastr.error($translate.instant('get.namespaces.error.msg', {error: getError(error)}));
+            });
+    };
+
+    const initResourceInfo = () => {
+        if ($scope.resourceInfo) {
+            return;
         }
-    });
+        $scope.resourceInfo = new ResourceInfo();
+        initResourceReference($scope.resourceInfo);
```

Comment:
This initResourceReference function doesn't accept any arguments.

Reply:
Fixed.